services/order.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+
+from db.models import Order, Ticket
+
+
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: Optional[str] = None) -> Order:
+    with transaction.atomic():
+        new_order = Order.objects.create(
+            user=get_user_model().objects.get(username=username)
+        )
+        if date:
+            new_order.created_at = date
+        for ticket_data in tickets:
+            Ticket.objects.create(
+                order=new_order,
+                movie_session_id=ticket_data["movie_session"],
+                row=ticket_data["row"],
+                seat=ticket_data["seat"]
+            )
+        new_order.save()
+        return new_order
+
+
+def get_orders(username: Optional[str] = None) -> Order:
```

Comment:
invalid typehint

Reply:
Why?

Reply:
Order.objects.filter() and Order.objects.all() are QuerySet

Reply:
It is not correct type hinting. You can test your code and check what will return .filter and .all
Or you can read documentation